### PR TITLE
Adding nektar++ package

### PR DESCRIPTION
**Pull request by mdavezac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nektar/package.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    version('4.3.0', '80dc0f3cb01a36e1a688bcc6d1cec561')
+    version('4.2.0', 'fee3df0e000a7257af1e9a83d5c461b3')
+    version('4.1.0', 'a2dc4e37849d96261b5107abdff3a088')
+    version('4.0.1', 'fcfc7b2771bfa310342b44f363782313')
+    version('4.0.0', '9a70068d4122781539bd7e3047a0eb2b')
+    version('3.4.0', '72d43a48e8c9b0085bcb025e4fc377e2')
+    version('3.3.0', '7b4aecd4dc4c8b1097d9626d684cecd0')
+    version('3.2.0', 'ba72e702a10b51fbcc5ed96b4fcfed67')
+    version('3.1.0', '1d03860ebdbdcdc1affad47897a398d9')
+    version('3.0.1', 'fb6d5a02f5c0bea62d611d2d29447c08')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nektar/package.py]: Do you actually need all of these versions? Have they even been tested? I only add the latest version when I write a new package.

File: var/spack/repos/builtin/packages/nektar/package.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    depends_on('mpi', when='+mpi')
+    depends_on('fftw@3.0: +mpi', when="+mpi+fftw")
+    depends_on('fftw@3.0: -mpi', when="-mpi+fftw")
+    depends_on('arpack-ng +mpi', when="+arpack+mpi")
+    depends_on('arpack-ng ~mpi', when="+arpack~mpi")
+    depends_on('hdf5 +mpi +hl', when="+mpi+hdf5")
+    depends_on('scotch -mpi -metis', when="-mpi+scotch")
+    depends_on('scotch +mpi -metis', when="+mpi+scotch")
+
+    conflicts("+hdf5", when="-mpi")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nektar/package.py]: Why is this the case? Maybe you can add a `msg` to the `conflicts` to explain.

Comment (healther) [reply var/spack/repos/builtin/packages/nektar/package.py]: Does `-variantname` work? I thought they were deprecated in favour of a consistent `~variantname`?

File: var/spack/repos/builtin/packages/nektar/package.py (hunk 1 above)
Comment (mdavezac) [reply var/spack/repos/builtin/packages/nektar/package.py]: Allright

File: var/spack/repos/builtin/packages/nektar/package.py (hunk 2 above)
Comment (mdavezac) [reply var/spack/repos/builtin/packages/nektar/package.py]: The reason is that nektar packages it's own version of metis.
I'll add a conflict and change "-" to "~".

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/nektar/package.py]: `-variant` and `~variant` should both work. Neither is deprecated as far as I know.

**Comment by Zehvogel** [comment]
thx for the work, I had already started to work on a nektar package on my own, but did not have the time to finish :disappointed:
I will test this

**Comment by Zehvogel** [comment]
I did not really have time to test it but @oliver-pola managed to compile it with everything on but scotch. Maybe he can comment :)

**Pull request merged**